pkg/inventory/types.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package inventory
+
+import "k8s.io/apimachinery/pkg/types"
+
+// Inventory represents the inventory object in memory.
+// Inventory is currently only used for in-memory storage and not serialized to
+// disk or to the API server.
+// TODO: Replace InventoryInfo with Inventory.TypeMeta & Inventory.ObjectMeta
+// TODO: Replace object.ObjMetadataSet in Storage interface with Inventory.Spec
+type Inventory struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   InventorySpec   `json:"spec,omitempty"`
+	Status InventoryStatus `json:"status,omitempty"`
+}
+
+// InventorySpec is the specification of the desired/expected inventory state.
+type InventorySpec struct {
+	Objects []ObjectReference `json:"objects,omitempty"`
+}
+
+// InventoryStatus is the status of the current/last-known inventory state.
+type InventoryStatus struct {
+	Objects []ObjectStatus `json:"objects,omitempty"`
+}
+
+// ObjectReference is a reference to a KRM resource by name and kind.
+//
+// Kubernetes only stores one API Version for each Kind at any given time,
+// so version is not used when referencing objects.
+type ObjectReference struct {
```

Comment:
This `ObjectReference` looks like a duplidate of `ObjMetadata`. Why do we need it?

Reply:
See PR comment: https://github.com/kubernetes-sigs/cli-utils/pull/531#issuecomment-1028511879